includes/tracking-events.php

```diff
@@ -0,0 +1,120 @@
+        }
+        
+        // if visitor is viewing product category
+        if (!$this->single_item_tracked && is_product_category()) {
+            $category = get_queried_object();
+            $this->put_event_in_queue(
+                "window.metrilo.viewCategory('"
+                . $category->term_id
+                . "');"
+                
+            );
+            $this->single_item_tracked = true;
+        }
+    
+        // if visitor is viewing search page
+        if (!$this->single_item_tracked && is_search()) {
+            $search = get_queried_object();
+            $this->put_event_in_queue(
+                "window.metrilo.search('"
+                . get_search_query()
+                . "', '"
+                . get_search_link()
+                . "');"
+        
+            );
+            $this->single_item_tracked = true;
+        }
+        
+        // if visitor is viewing shopping cart page
+        if(!$this->single_item_tracked && is_cart()){
+            $this->put_event_in_queue(
+                "window.metrilo.customEvent('view_cart');"
+            );
+            $this->single_item_tracked = true;
+        }
+        // if visitor is anywhere in the checkout process
+        if (!$this->single_item_tracked && is_order_received_page()) {
+            $this->put_event_in_queue(
+                "window.metrilo.viewPage('"
+                . get_permalink()
+                . "', "
+                . json_encode(['name' => get_the_title()])
+                . ");"
+            );
+            $this->single_item_tracked = true;
+            
+        } elseif (
+            !$this->single_item_tracked &&
+            function_exists('is_checkout_pay_page') &&
+            is_checkout_pay_page()
+        ) {
+            $this->put_event_in_queue(
+                "window.metrilo.customEvent('checkout_payment');"
+            );
+            $this->single_item_tracked = true;
+        } elseif (!$this->single_item_tracked && is_checkout()) {
+            $this->put_event_in_queue(
+                "window.metrilo.checkout();"
+            );
+            $this->single_item_tracked = true;
+        }
+    }
+    //
+    //  ** GENERIC WordPress tracking - doesn't require WooCommerce in order to work **//
+    //
+    //  if visitor is viewing homepage or any text page
+    if (!$this->single_item_tracked && is_front_page()) {
+        $this->put_event_in_queue(
+            "window.metrilo.viewPage('"
+            . get_permalink()
+            . "', "
+            . json_encode(['name' => get_the_title()])
+            . ");"
+        );
+        $this->single_item_tracked = true;
+    } elseif (!$this->single_item_tracked && is_page()) {
+        $this->put_event_in_queue(
+            "window.metrilo.viewPage('"
+            . get_permalink()
+            . "', "
+            . json_encode(['name' => get_the_title()])
+            . ");"
+        );
+        $this->single_item_tracked = true;
+    }
```

Comment:
The body of the `if` and the body of the `elseif` are exactly the same. The body of this `if` also: https://github.com/Metrilo/metrilo-woocommerce-integration/pull/49/files#diff-2a5043f710b4d68725df2206a7ece72331232be236e38fffcd93a788d8c82cdcR50